components/constituency_lookup/ConstituencyLookup.tsx

```diff
@@ -97,9 +97,9 @@ const throttledApi = async (
   postcode: string,
   addressSlug?: string,
 ): Promise<ConstituencyLookupResponse | null> => {
-  //TODO handle address lookups in the cache if/when we use DemoClub API
-  if (lookupCache.hasOwnProperty(postcode)) {
-    const cached = await lookupCache[postcode];
+  addressSlug = addressSlug || "";
+  if (!addressSlug && lookupCache.hasOwnProperty(postcode)) {
+    const cached = await lookupCache[postcode + addressSlug];
```

Comment:
Might suggest having a separator, so the cache key is something like `${postcode}/${addressSlug}`.

Also suggest defining it in a const so if we do change it, easier to just change one place without accidentally missing one, so `const cacheKey = `${postcode}/${addressSlug}`` or similar